packages/hardhat-core/src/internal/core/config/config-resolution.ts

```diff
@@ -179,7 +186,9 @@ function resolveHardhatNetworkConfig(
 
   const blockGasLimit =
     hardhatNetworkConfig.blockGasLimit ??
-    clonedDefaultHardhatNetworkParams.blockGasLimit;
+    (hardforkGte(getHardforkName(hardfork), HardforkName.OSAKA)
```

Comment:
The Osaka hardfork introduced a transaction gas cap. This makes the default block gas limit dependant on the hardfork. When the hardfork is post-Osaka, we use the transaction gas cap.

Reply:
For reference, geth provides a config option to disable the check for the transaction gas limit. By default, geth still runs the checks, but a user can override it.

Reply:
Handled by EDR in https://github.com/NomicFoundation/edr/pull/1200